testsupport/wait/member.go

```diff
@@ -1085,6 +1085,9 @@ func (a *MemberAwaitility) WaitForUser(name string) (*userv1.User, error) {
 			}
 			return false, err
 		}
+		if _, exists := user.Labels[toolchainv1alpha1.ProviderLabelKey]; !exists {
+			return false, nil
+		}
```

Comment:
now I understand why you added the label also to the preexisting user and identity.
I would make this check either a criteria for the wait function or move the check out of this block and verify the existence of the label here: 
https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/user_assertions.go#L74-L90

Reply:
Tempted to go with checking the existing of label in user assertions because its easier, but i believe adding a wait criteria is more consistent with the existing codebase. So going ahead with that.